Swiftification/Dictionary.swift

```diff
@@ -56,6 +56,17 @@ public extension Dictionary {
         return result
     }
     
+    /// Unions the other dictionaries with `self`. If more than
+    /// one dictionary has the same key, it will take on the rightmost
+    /// dictionary's value.
+    mutating func formUnion(first: Dictionary, _ rest: Dictionary...) {
```

Comment:
I lifted this from the names of the functions on Set in Swift 3. Apparently there's been a lot of discussion about them on mailing lists, but this is what they ended on. For now.